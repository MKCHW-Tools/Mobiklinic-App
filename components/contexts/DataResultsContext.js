import React, { createContext, useState } from 'react';

const DataResultsContext = createContext();

export const DataResultsProvider = ({ children }) => {
  const [dataResults, setDataResults] = useState('');
  const [benData, setBenData] = useState([]);
  const [userLog, setUserLog] = useState('');
  const [patientId, setPatientId] = useState('');
  const [userNames, setUserNames] = useState('');


  const updateDataResults = newDataResults => {
    setDataResults(newDataResults);
  };

  const updateBenData = newBenData => {
    setBenData(newBenData);
  };

  const updateUserLog = newUserLog => {
    setUserLog(newUserLog);
  };

  const updatePatientVac = newPatientVac => {
    setPatientVac(newPatientVac);
  };

  const updateUserNames = newUserNames => {
    setUserNames(newUserNames);
  };

  return (
    <DataResultsContext.Provider
      value={{
        dataResults,
        updateDataResults,
        benData,
        updateBenData,
        userLog,
        updateUserLog,
        patientId,
        setPatientId,
        userNames,
        updateUserNames
      }}
    >
      {children}
    </DataResultsContext.Provider>
  );
};

export default DataResultsContext;