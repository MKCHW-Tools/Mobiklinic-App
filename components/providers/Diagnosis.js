import * as React from 'react';

const DiagnosisContext = React.createContext();

const DiagnosisProvider = ({children}) => {
  const [followups, setFollowups] = React.useState([]);
  const [followup, setFollowup] = React.useState({});
  const [diagnoses, setDiagnoses] = React.useState([]);
  const [diagnosis, setDiagnosis] = React.useState({});

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const initial = React.useMemo(() => ({
    handleFollowup: () => {},
    handleFollowups: () => {},
    handleDiagnoses: async data => {},
    handleDiagnosis: () => {},
  }));

  return (
    <DiagnosisContext.Provider
      value={{
        diagnoses,
        followups,
        setDiagnoses,
        setFollowups,
      }}>
      {children}
    </DiagnosisContext.Provider>
  );
};

export {DiagnosisContext, DiagnosisProvider};
