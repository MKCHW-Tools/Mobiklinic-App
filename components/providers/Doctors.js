import * as React from 'react';

const DoctorsContext = React.createContext();

const DoctorsProvider = ({children}) => {
  const [doctors, setDoctors] = React.useState([]);

  return (
    <DoctorsContext.Provider
      value={{
        doctors,
        setDoctors,
      }}>
      {children}
    </DoctorsContext.Provider>
  );
};

export {DoctorsContext, DoctorsProvider};
