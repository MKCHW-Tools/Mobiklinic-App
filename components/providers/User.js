import * as React from "react";

const UserContext = React.createContext();

const UserProvider = ({ children }) => {
	const [accessToken, setAccessToken] = React.useState("");
	const [user, setUser] = React.useState({});
	const [isRegistering, setIsRegistering] = React.useState("no");

	return (
		<UserContext.Provider
			value={{
				accessToken,
				isRegistering,
				user,
				setUser,
				setAccessToken,
				setIsRegistering,
			}}>
			{children}
		</UserContext.Provider>
	);
};

export { UserProvider, UserContext };
