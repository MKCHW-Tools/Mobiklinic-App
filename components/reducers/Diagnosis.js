
export const initialState = {
    diagnoses:[],
    diagnosis:{},
    followup:{}
}

export const reducer = (state, action) => {
    switch(action.type) {
        case 'NEW_DIAGNOSIS':
            return {
                ...state,
                diagnosis
            }
        case 'FOLLOW_UP':
            return {
                ...state,
                followup
            }
        case 'DIAGNOSES':
            return {
                ...state,
                diagnoses
            }
        default:
            return state
    }
}