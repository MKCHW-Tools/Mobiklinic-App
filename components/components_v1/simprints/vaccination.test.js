
import React from 'react';
import { render, fireEvent, waitFor, act } from '@testing-library/react-native';
import { DiagnosisContext } from '../../providers/Diagnosis';
import DataResultsContext from '../../contexts/DataResultsContext';
import PatientData from '../../simprints/vaccination';
import { Alert } from 'react-native';
import fetchMock from 'jest-fetch-mock';


// Configure fetch to use the mock implementation
fetchMock.enableMocks();

// Utility function to prepare the date to be accepted by DatePicker's onChange method
const createDateTimeSetEvtParams = (date,) => {
    return [
        {
            type: 'set',
            nativeEvent: {
                timestamp: date.getTime(),
            },
        },
        date,
    ];
};

describe(' Vaccination - PatientData', () => {

    beforeEach(() => {
        fetchMock.resetMocks(); // Reset fetch mocks before each test
    });



    //Test that component renders correctly
    it('should render correctly', () => {
        // Arrange
        const navigation = { navigate: jest.fn() };
        const route = {};
        const diagnosisContext = { diagnoses: [] };
        const dataResultsContext = { dataResults: '', userLog: 'userLog', patientId: '', setPatientId: jest.fn() };
        const fetchSpy = jest.spyOn(global, 'fetch');
        fetchSpy.mockResolvedValueOnce({ ok: true, json: jest.fn() });
        const alertSpy = jest.spyOn(Alert, 'alert');

        // Act
        const screen = render(
            <DiagnosisContext.Provider value={diagnosisContext}>
                <DataResultsContext.Provider value={dataResultsContext}>
                    <PatientData navigation={navigation} route={route} />
                </DataResultsContext.Provider>
            </DiagnosisContext.Provider>
        );

        // Assert
        expect(screen).toMatchSnapshot();

        //test some elements are rendered
        expect(screen.getByText(' VACCINATION DATA')).toBeTruthy();
        expect(screen.getByText('Submit')).toBeTruthy();
        expect(screen.getByText('Vaccine Name:')).toBeTruthy();
        expect(screen.getByText(' Card number:')).toBeTruthy();
        expect(screen.getByText('Dose:')).toBeTruthy();
        expect(screen.getByText(' Date for Vaccination:')).toBeTruthy();
        expect(screen.getByText('Site Administered:')).toBeTruthy();
        expect(screen.getByText('Facility:')).toBeTruthy();
        expect(screen.getByText('Date for Next Dose:')).toBeTruthy();


    });

    it('handles date picker correctly', () => {
        const route = {};
        const navigation = { navigate: jest.fn(), goBack: jest.fn() };

        const { getByText, getByTestId, queryByTestId, queryByText } = render(
            <DiagnosisContext.Provider value={{ diagnoses: [] }}>
                <DataResultsContext.Provider value={{ dataResults: '', userLog: 'userLog', patientId: '', setPatientId: jest.fn() }}>
                    < PatientData navigation={navigation} route={route} />
                </DataResultsContext.Provider>
            </DiagnosisContext.Provider>);

        // Initially, date picker should not be visible
        expect(queryByTestId('dateOfVaccination')).toBeFalsy();

        // Simulate a button click to show the date picker
        fireEvent.press(getByTestId('showVaccDatePicker'));

        // Now, date picker should be visible
        expect(queryByTestId('dateOfVaccination')).toBeTruthy();

    });


    // Tests that the form can be submitted when all required fields are filled
    it('should submit form when all required fields are filled', () => {
        // Arrange
        const navigation = { navigate: jest.fn() };
        const route = {};
        const diagnosisContext = { diagnoses: [] };
        const dataResultsContext = { dataResults: '', userLog: 'userLog', patientId: '13', setPatientId: jest.fn() };

        // render the component
        const { getByText, getByTestId } = render(
            <DiagnosisContext.Provider value={diagnosisContext}>
                <DataResultsContext.Provider value={dataResultsContext}>
                    <PatientData navigation={navigation} route={route} />
                </DataResultsContext.Provider>
            </DiagnosisContext.Provider>
        );

        // Mock alert
        // Mock the Alert.alert function
        const alertSpy = jest.spyOn(Alert, 'alert').mockImplementation((title, message) => {

        });


        /**
        *Required Fields
        *  vaccineName
        *  dose
        *  dateOfVaccination
        *  siteAdministered
        */

        //fill in the form
        // Simulate a button click to show the date picker
        fireEvent.press(getByTestId('showVaccDatePicker'));
        const vaccineName = getByTestId('vaccineName');
        const dose = getByTestId('dose');
        const dateOfVaccination = getByTestId('dateOfVaccination');
        const siteAdministered = getByTestId('siteAdministered');
        const submitButton = getByText('Submit');


        // Simulate selecting a value
        fireEvent(vaccineName, 'onValueChange', "pfizer");
        fireEvent(dose, 'onValueChange', "1st");
        fireEvent(siteAdministered, 'onValueChange', "Left Arm");


        // Select a date from the date picker
        // Generate new date
        let date = new Date()

        // Fire the onChange Event
        fireEvent(
            getByTestId('dateOfVaccination'),
            'onChange',
            ...createDateTimeSetEvtParams(date),
        );

        // Mock a successful response from the server
        fetchMock.mockResponseOnce(JSON.stringify({ id: 1 }), { status: 200 });

        // Act
        fireEvent.press(submitButton);

        // Assert
        // Verify that the fetch request was made with the expected data
        expect(fetchMock).toHaveBeenCalledWith(
            `https://mobi-be-production.up.railway.app/${dataResultsContext.patientId}/vaccinations`,
            {
                method: 'POST',
                body: JSON.stringify({
                    vaccineName: 'pfizer',
                    dose: '1st',
                    units: '',
                    dateOfVaccination: createDateTimeSetEvtParams(date)[1],
                    dateForNextDose: '',
                    siteAdministered: 'Left Arm',
                    facility: '',
                }),
                headers: {
                    'Content-type': 'application/json; charset=UTF-8',
                    Accept: 'application/json',
                },
            }
        );
    });

    // Tests that an error message is displayed when an invalid vaccine name is selected
    it('should display error message when not all required fields are filled', () => {
        const navigation = { navigate: jest.fn() };
        const route = {};
        const diagnosisContext = { diagnoses: [] };
        const dataResultsContext = { dataResults: '', userLog: 'userLog', patientId: '13', setPatientId: jest.fn() };

        // render the component
        const { getByText, getByTestId } = render(
            <DiagnosisContext.Provider value={diagnosisContext}>
                <DataResultsContext.Provider value={dataResultsContext}>
                    <PatientData navigation={navigation} route={route} />
                </DataResultsContext.Provider>
            </DiagnosisContext.Provider>
        );

        // Mock alert
        // Mock the Alert.alert function
        const alertSpy = jest.spyOn(Alert, 'alert').mockImplementation((title, message) => {

        });


        /**
         *Required Fields
         *  vaccineName
         *  dose
         *  dateOfVaccination
         *  siteAdministered
         */

        //fill in the form
        const vaccineName = getByTestId('vaccineName');
        const dose = getByTestId('dose');
        const siteAdministered = getByTestId('siteAdministered');
        const submitButton = getByText('Submit');


        // Simulate selecting a value
        fireEvent(vaccineName, 'onValueChange', "");
        fireEvent(dose, 'onValueChange', "");
        fireEvent(siteAdministered, 'onValueChange', "Left Arm");

        // Act
        fireEvent.press(submitButton);

        // Assert
        // Verify that the fetch request was made with the expected data
        expect(fetchMock).not.toHaveBeenCalledWith(
            `https://mobi-be-production.up.railway.app/${dataResultsContext.patientId}/vaccinations`,
            {
                method: 'POST',
                body: JSON.stringify({
                    vaccineName: 'pfizer',
                    dose: '1st',
                    units: '',
                    dateOfVaccination: '',
                    dateForNextDose: '',
                    siteAdministered: 'Left Arm',
                    facility: '',
                }),
                headers: {
                    'Content-type': 'application/json; charset=UTF-8',
                    Accept: 'application/json',
                },
            }
        );

        // Assert
        expect(alertSpy).toHaveBeenCalledWith('Error', 'Please fill in all required fields');
    });


});
