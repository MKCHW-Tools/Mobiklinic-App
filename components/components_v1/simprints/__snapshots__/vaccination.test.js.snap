// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Vaccination - PatientData should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255,255,255,0.6)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomColor": "#ccc",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "height": 35,
          "justifyContent": "center",
          "marginHorizontal": 4,
          "opacity": 1,
          "width": 35,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#000",
              "fontSize": 25,
            },
            undefined,
            {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ÔÑê
      </Text>
    </View>
    <Text
      style={
        [
          {
            "alignItems": "center",
            "color": undefined,
            "flex": 2,
          },
          {
            "alignItems": "center",
            "color": "#000",
            "flexGrow": 1,
            "fontWeight": "bold",
          },
        ]
      }
    >
      VACCINATION DATA
    </Text>
  </View>
  <RCTScrollView
    style={
      {
        "flex": 2,
        "padding": 30,
      }
    }
  >
    <View>
      <View
        placeholderTextColor="rgba(0,0,0,0.7)"
        style={
          {
            "alignItems": "center",
            "borderColor": "#ccc",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 15,
          }
        }
        testID="vaccineNameParent"
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "medium",
              "marginLeft": 5,
              "marginRight": 5,
            }
          }
        >
          Vaccine Name:
        </Text>
        <View
          style={
            [
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 13,
                "fontWeight": "medium",
                "justifyContent": "center",
              },
              {
                "color": "#000",
              },
            ]
          }
        >
          <RNCPicker
            items={
              [
                {
                  "label": "",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "",
                },
                {
                  "label": "Pfizer-BioNTech",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "pfizer",
                },
                {
                  "label": "Moderna",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "moderna",
                },
                {
                  "label": "Johnson & Johnson",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "jnj",
                },
                {
                  "label": "AstraZeneca",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "astrazeneca",
                },
                {
                  "label": "Sinovac",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "sinovac",
                },
                {
                  "label": "Sinopharm",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "sinopharm",
                },
                {
                  "label": "Covishield",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "covishield",
                },
                {
                  "label": "Sputnik V",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "sputnik",
                },
                {
                  "label": "Hepatitis B Vaccine",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "hepatitisB",
                },
                {
                  "label": "Polio Vaccine",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "polio",
                },
                {
                  "label": "Tetanus, Diphtheria, and Pertussis Vaccine",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "tdap",
                },
                {
                  "label": "Pneumococcal Vaccine",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "pneumococcal",
                },
                {
                  "label": "Rotavirus Vaccine",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "rotavirus",
                },
                {
                  "label": "Influenza Vaccine",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "influenza",
                },
              ]
            }
            numberOfLines={1}
            onChange={[Function]}
            selectedIndex={0}
            style={
              [
                {
                  "height": 216,
                },
                undefined,
              ]
            }
            testID="vaccineName"
          />
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#ccc",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "medium",
              "marginLeft": 5,
              "marginRight": 5,
            }
          }
        >
          Card number:
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter vaccine card number"
          placeholderTextColor="#ccc"
          style={
            [
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 13,
                "fontWeight": "medium",
                "justifyContent": "center",
              },
              {
                "color": "#000",
              },
            ]
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#ccc",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "medium",
              "marginLeft": 5,
              "marginRight": 5,
            }
          }
        >
          Date for Vaccination:
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "#ccc",
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
          testID="showVaccDatePicker"
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 14,
                "marginLeft": 5,
                "paddingVertical": 12,
              }
            }
          >
            Click to add date
          </Text>
        </View>
      </View>
      <View
        placeholderTextColor="rgba(0,0,0,0.7)"
        style={
          {
            "alignItems": "center",
            "borderColor": "#ccc",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "medium",
              "marginLeft": 5,
              "marginRight": 5,
            }
          }
        >
          Dose:
        </Text>
        <View
          style={
            [
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 13,
                "fontWeight": "medium",
                "justifyContent": "center",
              },
              {
                "color": "#000",
              },
            ]
          }
        >
          <RNCPicker
            items={
              [
                {
                  "label": "",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "",
                },
                {
                  "label": "1st",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "1st",
                },
                {
                  "label": "2nd",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "2nd",
                },
                {
                  "label": "3rd",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "3rd",
                },
                {
                  "label": "4th",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "4th",
                },
              ]
            }
            numberOfLines={1}
            onChange={[Function]}
            selectedIndex={0}
            style={
              [
                {
                  "height": 216,
                },
                undefined,
              ]
            }
            testID="dose"
          />
        </View>
      </View>
      <View
        placeholderTextColor="rgba(0,0,0,0.7)"
        style={
          {
            "alignItems": "center",
            "borderColor": "#ccc",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "medium",
              "marginLeft": 5,
              "marginRight": 5,
            }
          }
        >
          Site Administered:
        </Text>
        <View
          style={
            {
              "color": "#000",
              "flex": 1,
              "fontSize": 13,
              "fontWeight": "medium",
              "justifyContent": "center",
            }
          }
        >
          <RNCPicker
            items={
              [
                {
                  "label": "",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "",
                },
                {
                  "label": "Left Upper Arm",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Left Upper Arm",
                },
                {
                  "label": "Right Upper Arm",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Right Upper Arm",
                },
                {
                  "label": "Left Upper Thigh",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Ventrogluteal Muscle (Hip)",
                },
                {
                  "label": "Mouth",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Mouth",
                },
                {
                  "label": "Buttocks",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Buttocks",
                },
                {
                  "label": "Right Upper Thigh",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Anterolateral Thigh (Front of the Thigh)",
                },
              ]
            }
            numberOfLines={1}
            onChange={[Function]}
            selectedIndex={0}
            style={
              [
                {
                  "height": 216,
                },
                undefined,
              ]
            }
            testID="siteAdministered"
          />
        </View>
      </View>
      <View
        placeholderTextColor="rgba(0,0,0,0.7)"
        style={
          {
            "alignItems": "center",
            "borderColor": "#ccc",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "medium",
              "marginLeft": 5,
              "marginRight": 5,
            }
          }
        >
          Facility:
        </Text>
        <View
          style={
            [
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 13,
                "fontWeight": "medium",
                "justifyContent": "center",
              },
              {
                "color": "#000",
              },
            ]
          }
        >
          <RNCPicker
            items={
              [
                {
                  "label": "",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "",
                },
                {
                  "label": "Buikwe Hospital",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Buikwe Hospital",
                },
                {
                  "label": "Mulago Hospital",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Mulago Hospital",
                },
                {
                  "label": "Makonge Health Center III",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Makonge Health Center III",
                },
                {
                  "label": "Makonge Health Center III",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Makonge Health Center III",
                },
                {
                  "label": "Makindu Health Center III",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Makindu Health Center III",
                },
                {
                  "label": "Kisungu Health Center III",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Kisungu Health Center III",
                },
                {
                  "label": "Najjembe Health Center III",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Najjemebe Health Center III",
                },
                {
                  "label": "Kawolo Hospital",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "Kawolo Hospital",
                },
              ]
            }
            numberOfLines={1}
            onChange={[Function]}
            selectedIndex={0}
            style={
              [
                {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#ccc",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "medium",
              "marginLeft": 5,
              "marginRight": 5,
            }
          }
        >
          Date for Next Dose:
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "#ccc",
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 14,
                "marginLeft": 5,
                "paddingVertical": 12,
              }
            }
          >
            Click to add date
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#000",
            "borderRadius": 10,
            "marginBottom": 10,
            "marginTop": 10,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
